src/NzbDrone.Common/Disk/DiskProviderBase.cs

```diff
@@ -109,6 +109,23 @@ public bool FileExists(string path, StringComparison stringComparison)
             }
         }
 
+        public bool FolderWritable(string path)
+        {
+            Ensure.That(path, () => path).IsValidPath();
+
+            try
+            {
+                var testPath = Path.Combine(path, "drone_test.txt");
+                File.WriteAllText(testPath, DateTime.Now.ToString());
```

Comment:
I think we should make the text:
`this file was created to verify if {path} is writable. It should've been automatically deleted. Feel free to delete it.`